less_users/views.py

```diff
@@ -40,104 +47,196 @@ class ProfileView(LoginRequiredMixin, View):
+                "my_active": my_active,
+                "my_visible": my_visible,
+                "my_all_count": my_all_count,
+            },
+        )
 
     def post(self, request, **kwargs):
-        return redirect('my_challenges')
+        return redirect("my_challenges")
 
 
-def activate_view(request, pk):
-    """it checks if there exists active user_challenge for logged in user and chosen challenge;
-    if not- it creates new active user_challenge"""
-    challenge_id = int(request.POST.get('activate'))
-    user_challenge_exist = \
-        UserChallenge.objects.filter(user=request.user, challenge_id=challenge_id, is_active=True)
-    if len(user_challenge_exist) == 0:
-        user_new_challenge = UserChallenge.objects.create(user=request.user, challenge_id=challenge_id)
-        user_new_challenge.save()
-        messages.success(request, 'You activated new challenge!')
-        return HttpResponseRedirect(reverse('challenge_detail', args=[str(pk)]))
-    else:
-        messages.warning(request, 'You still have this challenge active!')
-        return redirect('my_challenges')
-
-
-def event_view(request, **kwargs):
-    """gain points when challenge is completed; stop challenge or detele challenge """
-    user_id = request.user
-    if 'done' in request.POST:
-        """should check if points can be added (depending on durration and frequncy in model"""
-        challenge_id = int(request.POST.get('done'))
-        points = Challenge.objects.get(id=challenge_id).points
-        this = UserChallenge.objects.get(user_id=user_id, challenge_id=challenge_id, is_active=True)
-        user_challenge_id = this.id
-        this.check_if_active()
-        if this.get_points == points and this.is_active:
-            new_log = Log.objects.create(user_challenge_id=user_challenge_id, points=points)
-            my_profile = Profile.objects.get(user_id=user_id)
-            my_profile.points += points
-            my_profile.save()
-            messages.success(request, f'You have got new points {points} for challenge: {this.challenge.name}!')
-        elif this.get_points == 0 or this.is_active == False:
-            messages.warning(request, f'You cannot get new points for this challenge yet: {this.challenge.name}!')
-    elif 'stop' in request.POST:
-        user_challenge_id = int(request.POST.get('stop'))
-        this = UserChallenge.objects.get(id=user_challenge_id)
-        this.is_active = False
-        this.save()
-        messages.info(request, f'You stopped challenge: {this.challenge.name}. You can activate the new one.')
-    elif 'delete' in request.POST:
-        user_challenge_id = int(request.POST.get('delete'))
-        this = UserChallenge.objects.get(id=user_challenge_id)
-        this.delete()
-        messages.warning(request, f'You have deleted {this.challenge.name} from your challenges')
-    return redirect('my_challenges')
+# return HttpResponseRedirect(reverse("my_challenges"), request.path_info)
```

Comment:
Wykomentowany kod

Reply:
ten kod to jest moja "notatka" na kolejny krok, którym miało być przeformatowanie kodu tak, żeby strona my_challenges nie ładowała się zawsze od nowa gdy POSTem określam jej "order_by"... ale rozumiem, że takie notatki to lepiej w notatniku niż w kodzie :)

Reply:
Zdecydowanie lepszym sposobem notowania następnych zadań jest dodawanie issuesów ;)

Reply:
o, nowa funkcjonalność do obadania w githubie